cmd/soroban-cli/src/commands/config/locator.rs

```diff
@@ -211,6 +220,78 @@ impl Args {
     pub fn remove_network(&self, name: &str) -> Result<(), Error> {
         KeyType::Network.remove(name, &self.config_dir()?)
     }
+
+    fn load_contract_from_alias(&self, alias: &str) -> Result<Option<alias::Data>, Error> {
+        let path = self.alias_path(alias)?;
+
+        if !path.exists() {
+            return Ok(None);
+        }
+
+        let content = fs::read_to_string(path)?;
+        let data: alias::Data = serde_json::from_str(&content).unwrap_or_default();
+
+        Ok(Some(data))
+    }
+
+    fn alias_path(&self, alias: &str) -> Result<PathBuf, Error> {
+        let file_name = format!("{alias}.json");
```

Comment:
IMO this is where the validation should happen. 
https://github.com/stellar/stellar-cli/blob/85ec1ee0a3fac79d09111544e894c14741e14e08/cmd/soroban-cli/src/commands/contract/deploy/wasm.rs#L136

Currently we only allow making an alias with deploy, but we could add a way to add one via a `contract alias` command.

Reply:
As mentioned in the original pr, we can't move the validation to a later step, otherwise the contract could be deployed with an invalid alias.

> Currently we only allow making an alias with deploy, but we could add a way to add one via a contract alias command.

I'm not so sure about adding yet another command to manage aliases right now. Let's see if people ask for it and then we can consider it.

Reply:
Yeah I guess for me deploying was still okay, but without another way to add the alias I get it. I guess I would prefer validation via clap: .g.
```rust
/// The alias that will be used to save the contract's id.
#[arg(long, value_parser = clap::builder::ValueParser::new(alias_validator))]
pub alias: Option<String>,
```

``` rust
fn alias_validator(s: &str) -> Result<String, String> {
    let regex = regex::Regex::new(r"^[a-zA-Z0-9_-]+$").unwrap();
    if regex.is_match(s) {
        Ok(s.to_string())
    } else {
        Err(...)
    }
}
```
This way if we moved alias to its own `Arg` struct it could fail at the parsing step. However, again not an issue until or if we add more commands.

Reply:
oh, i think that's a good change. I didn't do it initially because… because I never bothered to look at clap's docs. 😬🫠